src/stream/src/common/table/state_table.rs

```diff
@@ -630,18 +644,33 @@ impl<S: StateStore> StateTable<S> {
         self.epoch = Some(new_epoch);
     }
 
-    pub async fn commit(&mut self, new_epoch: EpochPair) -> StreamExecutorResult<()> {
+    pub async fn commit(
+        &mut self,
+        new_epoch: EpochPair,
+        watermark: Option<&ScalarImpl>,
+    ) -> StreamExecutorResult<()> {
         assert_eq!(self.epoch(), new_epoch.prev);
         let mem_table = std::mem::take(&mut self.mem_table).into_parts();
-        self.batch_write_rows(mem_table, new_epoch.prev).await?;
+        self.num_commits_since_last_clean += 1;
```

Comment:
Could we move this code to the method `batch_write_rows` and only calculate it when `watermark` is not None ?

Reply:
> Could we move this code to the method `batch_write_rows` and only calculate it when `watermark` is not None ?

fixed

Reply:
> Could we move this code to the method `batch_write_rows` and only calculate it when `watermark` is not None ?

this part logic has been moved into #6495 because of @jon-chuang 's suggestion in #6473